### PR TITLE
Detect hyphen in hard coded reference

### PR DESCRIPTION
**Pull request by simmartynas**
This is a:
- [ ] bug fix PR with no breaking changes
- [x] new functionality

## Link to Issue 
https://github.com/dbt-labs/dbt-project-evaluator/issues/417

## Description & motivation
In BigQuery, table references can contain - in them. Test find_all_hard_coded_references.sql does not detect this because regex (\w+) does not include the hyphen (-) character.

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [x] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [ ] DuckDB
    - [ ] Trino/Starburst
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

**Review by dave-connors-3 — APPROVED** [review]
thanks so much for your contribution @simmartynas

File: macros/find_all_hard_coded_references.sql [hunk 1]
```diff
@@ -164,48 +164,48 @@
...
                 (\.)
 
                 # sixth matching group
                 # 1 or 0 of (opening bracket, backtick, or quotation mark)
                 ([\[`\"\']?)
 
                 # seventh matching group
-                # at least 1 word character            
-                (\w+)
+                # at least 1 word character
```
Comment (graciegoheen) [inline macros/find_all_hard_coded_references.sql:191]: wouldn't this change this to "at least 1 word character followed by 0 or 1 dash followed by at least 1 word character?)

File: integration_tests/seeds/dag/test_fct_hard_coded_references.csv [hunk 2]
```diff
@@ -1,2 +1,2 @@
...
-fct_model_6,"""my_db"".""my_schema"".""my_table"", ""my_schema"".""raw_relation_5"", ""raw_relation_2"", 'my_db'.'my_schema'.'my_table', 'my_schema'.'raw_relation_5', 'raw_relation_4', [my_db].[my_schema].[my_table], [my_schema].[raw_relation_5], [raw_relation_3], `my_db`.`my_schema`.`my_table`, `my_schema`.`raw_relation_5`, `raw_relation_1`, my_db.my_schema.my_table, my_schema.raw_relation_5, {{ var(""my_table_reference"") }}, {{ var(""my_table_reference"", ""table_d"") }}, {{ var('my_table_reference') }}, {{ var('my_table_reference', 'table_d') }}"
\ No newline at end of file
+fct_model_6,"""my_db"".""my_schema"".""my_table"", ""my_schema"".""raw_relation_5"", ""raw_relation_2"", 'my_db'.'my_schema'.'my_table', 'my_schema'.'raw_relation_5', 'raw_relation_4', [my_db].[my_schema].[my_table], [my_schema].[raw_relation_5], [raw_relation_3], `my_db`.`my_schema`.`my_table`, `my_schema`.`raw_relation_5`, `project-name.dataset_name.table_name`, `raw_relation_1`, my_db.my_schema.my_table, my_schema.raw_relation_5, {{ var(""my_table_reference"") }}, {{ var(""my_table_reference"", ""table_d"") }}, {{ var('my_table_reference') }}, {{ var('my_table_reference', 'table_d') }}"
```
Comment (graciegoheen) [inline integration_tests/seeds/dag/test_fct_hard_coded_references.csv:2]: Is this only for the three argument case `project-name.dataset_name.table_name`?

Comment (graciegoheen) [reply integration_tests/seeds/dag/test_fct_hard_coded_references.csv:2]: below it looks like every instance of  `(\w+-?\w+)` was updated? if so, we would then want to test other iterations of `-`

**Pull request merged**